### PR TITLE
Reviewed: 1011-rules-for-reporting-org

### PR DESCRIPTION
**Pull request by PG-Momik**

### CODE REVIEW

File: .husky/pre-commit [hunk 1]
```diff
@@ -1,9 +1,10 @@
 #!/bin/sh
 . "$(dirname "$0")/_/husky.sh"
 
-php artisan test && \
+# php artisan test && \
```
Comment (man-git007) [inline .husky/pre-commit:4]: uncomment this code

File: app/XmlImporter/Foundation/Support/Factory/Validation.php [hunk 2]
```diff
@@ -535,4 +535,14 @@ protected function getErrors($element, $elementErrors, $activityId): array
 
         return $errors;
     }
+
+    public function mustMatchValidationForReportingOrg(): void
```
Comment (man-git007) [inline app/XmlImporter/Foundation/Support/Factory/Validation.php:539]: php doc missing

File: app/Console/Commands/FillReportingOrgInOrganizationLevel.php [hunk 3]
```diff
@@ -0,0 +1,102 @@
+<?php
+
+namespace App\Console\Commands;
```
Comment (man-git007) [inline app/Console/Commands/FillReportingOrgInOrganizationLevel.php:3]: add `strict type` declaration

File: app/Console/Commands/FillReportingOrgInOrganizationLevel.php [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+
+use App\IATI\Repositories\Activity\ActivityRepository;
+use App\IATI\Repositories\Organization\OrganizationRepository;
+use Illuminate\Console\Command;
+use Illuminate\Support\Facades\DB;
+
+class FillReportingOrgInOrganizationLevel extends Command
```
Comment (man-git007) [inline app/Console/Commands/FillReportingOrgInOrganizationLevel.php:10]: php doc missing

File: app/CsvImporter/Entities/Activity/Components/Elements/ReportingOrganization.php [hunk 5]
```diff
@@ -14,17 +14,21 @@
  */
 class ReportingOrganization extends Element
 {
+    /**
+     * @var
+     */
+    public $reportingOrg;
```
Comment (man-git007) [inline app/CsvImporter/Entities/Activity/Components/Elements/ReportingOrganization.php:20]: define a variable type like if it is array or string or integer.

`public $reportingOrg = [];` or `public $reportingOrg = '';`

File: app/CsvImporter/Entities/Activity/Components/Elements/ReportingOrganization.php [hunk 6]
```diff
@@ -14,17 +14,21 @@
...
+
+    /**
+     * @var
+     */
+    public $temp;
```
Comment (man-git007) [inline app/CsvImporter/Entities/Activity/Components/Elements/ReportingOrganization.php:25]: define variable type

File: app/IATI/Repositories/Activity/ActivityRepository.php [hunk 7]
```diff
@@ -457,4 +458,42 @@ public function getAllActivitiesToDownload($organizationId, $queryParams): objec
...
+     *
+     * @param $id
+     * @param $reportingOrg
+     *
+     * @return object|int
+     */
+    public function syncReportingOrg($id, $reportingOrg): object|int
+    {
+        $activitiesCount = $this->model->where('org_id', $id)->count();
+        if ($activitiesCount > 0) {
```
Comment (man-git007) [inline app/IATI/Repositories/Activity/ActivityRepository.php:473]: add new line

File: app/IATI/Services/ImportActivity/ImportCsvService.php [hunk 8]
```diff
@@ -202,6 +202,15 @@ public function create($activities): void
             $iati_identifier_text = $organizationIdentifier . '-' . Arr::get($activity, 'data.identifier.activity_identifier');
             $activity['data']['identifier']['iati_identifier_text'] = $iati_identifier_text;
 
+            if (count($activity['data']['reporting_organization']) > 1) {
+                $activity['data']['reporting_organization'][0]['secondary_reporter'] = '';
+                foreach ($activity['data']['reporting_organization'] as $key => $value) {
```
Comment (man-git007) [inline app/IATI/Services/ImportActivity/ImportCsvService.php:207]: add new line

File: resources/views/web/about.blade.php [hunk 9]
```diff
@@ -3,5 +3,65 @@
 @section('title', 'About IATI Publisher')
 
 @section('content')
-<about-page></about-page>
```
Comment (man-git007) [inline resources/views/web/about.blade.php:6]: remove this code

**Review by man-git007 — COMMENTED** [review]
Fix the comments added

File: app/Http/Requests/Activity/ReportingOrg/ReportingOrgRequest.php [hunk 10]
```diff
@@ -91,22 +133,72 @@ public function getWarningForReportingOrganization(array $formFields): array
...
+
             foreach ($narrativeMessages as $key => $item) {
                 $messages[$key] = $item;
             }
         }
 
         return $messages;
     }
+
+/**
```
Comment (man-git007) [inline app/Http/Requests/Activity/ReportingOrg/ReportingOrgRequest.php:165]: fix the spacing

File: app/Console/Commands/FillReportingOrgInOrganizationLevel.php [hunk 11]
```diff
@@ -0,0 +1,116 @@
...
+                $organization->updateOrInsert(
+                    ['id' => $organization->id],
+                    [
+                        'reporting_org' => json_encode($organization->reporting_org),
+                        'status'        => 'draft',
+                    ]
+                );
+                $organization = $organizations->find($organization->id);
+                $reportingOrg = $organization->reporting_org[0];
+
```
Comment (A4family) [inline app/Console/Commands/FillReportingOrgInOrganizationLevel.php:63]: Remove this Extra line

File: app/CsvImporter/Queue/Processor.php [hunk 12]
```diff
@@ -40,14 +40,14 @@ public function __construct(CsvReader $csvReader)
      *
      * @return void
      */
-    public function pushIntoQueue($file, $filename, $activityIdentifiers): void
+    public function pushIntoQueue($file, $filename, $activityIdentifiers, $organizationReportingOrg): void
     {
         $str = mb_convert_encoding(file_get_contents($file->getPathName()), 'UTF-8');
         file_put_contents($file->getPathName(), $str);
         $csv = Excel::toCollection(new CsvToArrayWithHeaders, $file)->first()->toArray();
```
Comment (A4family) [inline app/CsvImporter/Queue/Processor.php:48]: Remove this extra line

File: app/Http/Requests/Organization/ReportingOrg/ReportingOrgRequest.php [hunk 13]
```diff
@@ -40,13 +41,27 @@ public function messages(): array
...
+            $rules[$reportingOrganizationForm . '.type'] = [
+                'nullable',
+                sprintf('in:%s', implode(',', array_keys(getCodeList('OrganizationType', 'Organization')))),
+            ];
+            $rules[$reportingOrganizationForm . '.secondary_reporter'] = [
+                'nullable',
+                'in:0,1',
+            ];
 
             $narrativeRules = $this->getWarningForNarrative($reportingOrganization['narrative'], $reportingOrganizationForm);
```
Comment (A4family) [inline app/Http/Requests/Organization/ReportingOrg/ReportingOrgRequest.php:67]: remove this extra line

File: app/Http/Requests/Organization/ReportingOrg/ReportingOrgRequest.php [hunk 14]
```diff
@@ -73,6 +88,7 @@ public function getMessagesForReportingOrganization(array $formFields): array
             $reportingOrganizationForm = sprintf('reporting_org.%s', $reportingOrganizationIndex);
```
Comment (A4family) [inline app/Http/Requests/Organization/ReportingOrg/ReportingOrgRequest.php:89]: remove this extra line

File: app/IATI/Services/Organization/ReportingOrgService.php [hunk 15]
```diff
@@ -23,15 +24,22 @@ class ReportingOrgService
...
      * @param OrganizationRepository $organizationRepository
+     * @param ActivityRepository $activityRepository
      * @param ParentCollectionFormCreator $parentCollectionFormCreator
      */
-    public function __construct(OrganizationRepository $organizationRepository, ParentCollectionFormCreator $parentCollectionFormCreator)
+    public function __construct(OrganizationRepository $organizationRepository, ActivityRepository $activityRepository, ParentCollectionFormCreator $parentCollectionFormCreator)
     {
         $this->organizationRepository = $organizationRepository;
+        $this->activityRepository = $activityRepository;
         $this->parentCollectionFormCreator = $parentCollectionFormCreator;
```
Comment (Sanilblank) [inline app/IATI/Services/Organization/ReportingOrgService.php:43]: declare property above

**Review by sriza — APPROVED** [review]

**Pull request merged**